src/entities/Auth/Auth.validation.ts

```diff
@@ -0,0 +1,14 @@
+import Joi, { ValidationResult } from "joi";
+import AuthReq from './Auth.interface';
+
+const validateAuthReq = (authReq: AuthReq): ValidationResult => {
+    const schema = Joi.object({
+        // eslint-disable-next-line max-len
+        email: Joi.string().min(5).max(255).required().email().pattern(/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/), 
```

Comment:
```suggestion
        email: Joi.string()
        .min(5)
        .max(255)
        .required()
        .email()
        .pattern(/[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/), 
```

Reply:
Isn't `.email()` validation enough to ensure that this is email? Why adding `.pattern(...)`with regex?